torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -111,29 +122,73 @@ def create_subclass_meta(
 # a list of tensors that we would then need to concat together.
 # Instead, we specialize the logic for the inference vs. joint graph case.
 # NOTE: this function is hot, since we unwrap tensor subclass inputs at runtime
-def unwrap_tensor_subclasses(wrapped_args, *, is_joint_structure: bool):
-    def concat_inner_tensors_from_subclasses(xs):
-        xs_inner = []
-        for x in xs:
-            if is_traceable_wrapper_subclass(x):
-                xs_inner.extend(get_plain_tensors(typing.cast(Tensor, x)))
-            else:
-                xs_inner.append(x)
-        return xs_inner
 
-    if is_joint_structure:
-        assert isinstance(wrapped_args, tuple) and len(wrapped_args) == 2
-        assert isinstance(wrapped_args[0], (tuple, list)) and isinstance(
-            wrapped_args[1], (tuple, list)
-        )
-        unwrapped_args_fw = concat_inner_tensors_from_subclasses(wrapped_args[0])
-        unwrapped_args_tangents = concat_inner_tensors_from_subclasses(wrapped_args[1])
-        unwrapped_args = (unwrapped_args_fw, unwrapped_args_tangents)
-    else:
-        assert isinstance(wrapped_args, (list, tuple))
-        unwrapped_args_fw = concat_inner_tensors_from_subclasses(wrapped_args)
-        unwrapped_args = unwrapped_args_fw
-    return unwrapped_args
+
+def filter_symints(lst: Iterable[Union[int, SymInt]]) -> List[SymInt]:
+    return [s for s in lst if isinstance(s, SymInt) and not s.node.is_nested_int()]
```

Comment:
In particular, we might have to be a bit careful with how we use this function:

for any NJT graph inputs, we *will* want to add the `nested_int` as an extra graph arg (in case it is used in any compute in the graph

for any NJT graph outputs, we *wont* want to add their `nested_int` as an extra graph output (we want it to be ignored, because we can safely assume that the NJT output's inner `offsets` tensor will be an extra graph output that `NestedTensor.__tensor_unflatten__` will use instead)

Reply:
Some inductor tests are failing because we have nested ints as arguments to the forward/backward functions.